src/lib.rs

```diff
@@ -268,6 +270,7 @@ impl BarState {
             update_request: update_request.clone(),
             request_sender: self.request_sender.clone(),
             error_interval: Duration::from_secs(block_config.common.error_interval),
+            geolocator: Arc::new(Geolocator::new(self.config.geolocator.clone())),
```

Comment:
This creates a new instance of geolocator per each block, which prevents one block to share the cache with another block, if I understood the code correctly.

Reply:
Thanks, that was a good catch! I have updated the PR so that the `Config` holds the `Arc<Geolocator>` and then the reference is cloned into the `CommonApi`